Mathlib/NumberTheory/NumberField/Units.lean

```diff
@@ -525,6 +524,19 @@ instance : Module.Free ℤ (Additive ((𝓞 K)ˣ ⧸ (torsion K))) :=
 instance : Module.Finite ℤ (Additive ((𝓞 K)ˣ ⧸ (torsion K))) :=
   Module.Finite.equiv (unitLatticeEquiv K)
 
+instance : Module.Finite ℤ (Additive (𝓞 K)ˣ) := by
```

Comment:
Is this easier to prove by starting with [`Module.Finite.iff_addGroup_fg`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Finiteness.html#Module.Finite.iff_addGroup_fg)?

Reply:
At any rate I think you should provide s `Monoid.FG (Additive (𝓞 K)ˣ)` instance somehow.

Reply:
I added the instance for `Monoid.FG (𝓞 K)ˣ`.

> Is this easier to prove by starting with [Module.Finite.iff_addGroup_fg](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Finiteness.html#Module.Finite.iff_addGroup_fg) and [Monoid.fg_iff_add_fg](https://leanprover-community.github.io/mathlib4_docs/Mathlib/GroupTheory/Finiteness.html#Monoid.fg_iff_add_fg)?

The thing is that we don't have (as far as I can tell) a version of [Submodule.fg_of_fg_map_of_fg_inf_ker](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Finiteness.html#Submodule.fg_of_fg_map_of_fg_inf_ker) for commutative groups. It will be easy to add one (using the Submodule version since it's basically what I am doing) but then it's a bit of code duplication.

Reply:
Can you add a comment to that effect in the code? Something like:
```
-- note we prove the `Monoid.FG` in terms of this and not vice versa due to
-- no `Subgroup` version of `Submodule.fg_of_fg_map_of_fg_inf_ker` existing
```